ui/src/components/friends/friends_list/mod.rs

```diff
@@ -42,14 +90,16 @@ pub fn Friends(cx: Scope) -> Element {
                         },
                         sorted_friends.into_iter().map(|friend| {
                             let did = friend.did_key();
+                            let chat = state.read().get_chat_with_friend(&friend);
```

Comment:
unfortunately not. `state` doesn't need cloning because it's a `use_shared_state`.  the other variables have to go into a `move` closure and after that they can't be accessed again.